### PR TITLE
Add ignore user API support

### PR DESCRIPTION
**Pull request by turt2live**
General ticket:  https://github.com/vector-im/riot-web/issues/1490

* [x] Show which users are ignored in UserSettings (https://github.com/vector-im/riot-web/issues/1767)
* [x] Remove users from ignore list through UserSettings
* [x] (un)ignore people through MemberInfo
* [x] `/(un)ignore <user>`  slash command
* [x] Go back in time and hide events the user likely doesn't want to see
  * Intentionally don't re-sync to reduce pain for larger accounts

#### Requires
* https://github.com/matrix-org/matrix-js-sdk/pull/539
* https://github.com/vector-im/riot-web/pull/5042

### CODE REVIEW

File: src/components/structures/UserSettings.js [hunk 1]
```diff
@@ -228,6 +257,7 @@ module.exports = React.createClass({
         }
 
         this._refreshMediaDevices();
+        this._refreshIgnoredUsers();
```
Comment (ara4n) [inline src/components/structures/UserSettings.js]: I guess technically we might want to also _refreshIgnoreUsers() when we receive an ignoring event from the js-sdk.  But that's such an edge case i'm happy to ignore it for now :)

Comment (ara4n) [reply src/components/structures/UserSettings.js]: (although given we already have an onAction() here, surely it's just 3 lines to add in a:
```
case 'ignore_state_changed':
                this._refreshIgnoredUsers();
                break;
```
...which almost seems worth it.

File: src/components/views/rooms/MemberInfo.js [hunk 2]
```diff
@@ -225,6 +233,18 @@ module.exports = withMatrixClient(React.createClass({
         });
     },
 
+    onIgnoreToggle: function() {
+        const ignoredUsers = this.props.matrixClient.getIgnoredUsers();
+        if (this.state.isIgnoring) {
+            const index = ignoredUsers.indexOf(this.props.member.userId);
```
Comment (ara4n) [inline src/components/views/rooms/MemberInfo.js]: why not `matrixClient.isUserIgnored()`?

**Comment by ara4n** [comment]
this looks excellent - thank you!!! sorry it took us a few days to get to it; with lúke away, me stuck in funding hell and dave up to his eyeballs in perf work we're a bit undermanned.

**Comment by turt2live** [comment]
Thanks @ara4n! 2 days is pretty good compared to some of the other projects I contribute to (/me glares at 2015 PR)

**Comment by ara4n** [comment]
yeah, glad we don't have any like that... *glances at https://github.com/matrix-org/matrix-react-sdk/pull/588* :((

**Pull request merged**